### PR TITLE
fix(image_projection_based_fusion): add run length decoding for segmentation_pointcloud_fusion

### PR DESCRIPTION
**Pull request by badai-nguyen**
## Description
- Since the segmentation mask will be encoded before publish as  PR: https://github.com/autowarefoundation/autoware.universe/pull/7905, so this PR is to add corresponding decoder for that.
- This PR need merge after #7905 merged.
- Related issue: https://github.com/autowarefoundation/autoware.universe/issues/5594

## Related links

**Parent Issue:**

- Link

## How was this PR tested?

- Run `ros2 launch autoware_tensorrt_yolox yolox.launch.xml`  with `is_publish_color_mask: true`
- Run `ros2 launch autoware_image_projection_based_fusion segmentation_pointcloud_fusion.launch.xml` with `is_publish_debug_mask: true`
- Play rosbag (`input_bag` of an internal dataset) which contains `/sensing/camera/camera0/image_rect_color/compressed` and Pointcoud2 input topic for `/segmentation_pointcloud_fusion` node
- Confirmed the debug images by rqt that have the same segmentation result as below (Note: the colors are different because of different color map was used)  
image

## Notes for reviewers

None.

## Interface changes

None.

## Effects on system behavior

None.

### CODE REVIEW

File: perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp [hunk 1]
```diff
@@ -57,19 +60,23 @@ void SegmentPointCloudFusionNode::fuseOnSingleImage(
...
+    sensor_msgs::msg::Image::SharedPtr debug_mask_msg =
+      cv_bridge::CvImage(std_msgs::msg::Header(), "mono8", mask).toImageMsg();
+    debug_mask_msg->header = input_mask.header;
+    pub_debug_mask_ptr_.publish(debug_mask_msg);
   }
+  Eigen::Matrix4d projection;
+  projection << camera_info.p.at(0), camera_info.p.at(1), camera_info.p.at(2), camera_info.p.at(3),
+    camera_info.p.at(4), camera_info.p.at(5), camera_info.p.at(6), camera_info.p.at(7),
+    camera_info.p.at(8), camera_info.p.at(9), camera_info.p.at(10), camera_info.p.at(11), 0.0, 0.0,
+    0.0, 1.0;
```
Comment (kminoda) [inline perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:79]: Is this used somewhere?

Comment (badai-nguyen) [reply perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp]: @kminoda thank you for you comment. I my mistake when rebased branch. I fixed t https://github.com/autowarefoundation/autoware.universe/pull/7909/commits/b8abe57a82d8655e33bcb24e7ba2e9e2a69ef7fc

File: perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp [hunk 2]
```diff
@@ -16,6 +16,7 @@
 
 #include "autoware/image_projection_based_fusion/utils/geometry.hpp"
 #include "autoware/image_projection_based_fusion/utils/utils.hpp"
+#include "perception_utils/run_length_encoder.hpp"
```
Comment (kminoda) [inline perception/autoware_image_projection_based_fusion/src/segmentation_pointcloud_fusion/node.cpp:19]:
```suggestion
#include <perception_utils/run_length_encoder.hpp>
```

**Review by kminoda — COMMENTED** [review]
I added one small comment :pray: 

Also, please update `How was this PR tested?` section.

**Review by YoshiRi — APPROVED** [review]
LGTM

**Comment by badai-nguyen** [comment]
> I added one small comment 🙏
> 
> Also, please update `How was this PR tested?` section.

@kminoda thanks for your comment.
I updated `How was PR tested`. Could you check again. 
By the way, I also open PR for unit test to confirm the output after decoder the same with the input image here :https://github.com/autowarefoundation/autoware.universe/pull/8414

**Comment by badai-nguyen** [comment]
@kminoda Do you have other comments? Otherwise, I will merge this PR.

**Pull request merged**